src/cdk/collections/selection-model.ts

```diff
@@ -238,6 +233,21 @@ export class SelectionModel<T> {
   private _hasQueuedChanges() {
     return !!(this._deselectedToEmit.length || this._selectedToEmit.length);
   }
+
+  /** Returns a value that is comparable to inputValue by applying compareWith function, returns the same inputValue otherwise. */
+  private _getConcreteValue(inputValue: T): T {
+    let concreteValue = inputValue;
+    if (!this.compareWith) {
+      return concreteValue;
+    } else {
+      this._selection.forEach(selectedValue => {
+        if (this.compareWith?.call(undefined, inputValue, selectedValue)) {
+          concreteValue = selectedValue;
+        }
+      });
+      return concreteValue;
```

Comment:
Wouldn't we want to return the `inputValue` if there is no match?

Reply:
shouldn't change anything since we first assign `concreteValue  `to `inputValue `, edited for clarity.

Reply:
Ultimately, there is no need to create the `concreteValue` variable, we'll just `return` inside of the `for` loop in order to break it when there is a match.